### PR TITLE
Bugfix FXIOS-10740 workaround for crashing on non-unique Sites in the HistoryPanel

### PR DESCRIPTION
**Pull request by ih-codes**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-10740)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/23448)

## :bulb: Description
Temporary workaround for non-unique Sites in the HistoryPanel. We should get a GUID that is unique from application services later... for now just force a unique UUID for each history item (not idea for diffable data sources of course).

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Storage/Rust/RustPlaces.swift [hunk 1]
```diff
@@ -666,7 +666,10 @@ extension RustPlaces {
                     // as the title
                     title = info.url
                 }
-                let site = Site(url: info.url, title: title)
+                // FIXME: FXIOS-10740 Workaround for an iOS 18 HistoryPanel crash with diffable data sources until we can
+                // get a unique GUID from application services (should be sent in the payload with synced history items)
+                let hashValue = "\(info.url)_\(info.timestamp)".hashValue
```
Comment (ih-codes) [inline firefox-ios/Storage/Rust/RustPlaces.swift:671]: If we REALLY want to be confident this is unique, we should use `let hashValue = UUID().uuidString.hashValue`...

**Review by nbhasin2 — APPROVED** [review]
Lets see if we can incorporate GUID

**Comment by ih-codes** [comment]
I'm seeing some unit test failures, which I will address Monday.

**Review by PARAIPAN9 — APPROVED** [review]
Wonderful job, @ih-codes!!

**Comment by ih-codes** [comment]
@Mergifyio backport release/v134

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/HistoryTests.swift [hunk 2]
```diff
@@ -530,9 +530,6 @@ class HistoryTests: BaseTestCase {
     // https://mozilla.testrail.io/index.php?/cases/view/2306890
     // Smoketest
     func testDeleteHistoryEntryBySwiping() {
-        navigateToPage()
-        navigator.goto(LibraryPanel_History)
-        waitForExistence(app.cells.staticTexts["http://example.com/"])
```
Comment (ih-codes) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/HistoryTests.swift:535]: @clarmso I'm not really sure what's going on here, but this test visits 2 websites, but only deletes 1, and then expects the empty message to show.

Some of my work on this branch fixes the equality check for website duplicates. My guess is that both sites were being deleted before, which was why this test passed?? 🤔

Comment (ih-codes) [reply firefox-ios/firefox-ios-tests/Tests/XCUITests/HistoryTests.swift:535]: I just tried running this test on main and even though the test visits 2 websites, only 1 appears in the history, so I must've fixed another bug in this PR!

Here is the main run:
https://github.com/user-attachments/assets/5ded0f66-6e87-4e0a-84c2-5c041dc8ae51

Comment (ih-codes) [reply firefox-ios/firefox-ios-tests/Tests/XCUITests/HistoryTests.swift:535]: P.S. Here is a video of the run with my fixes on this branch:

https://github.com/user-attachments/assets/11169972-46cc-4bf1-97d2-2405117767aa

**Review by mdotb-moz — APPROVED** [review]
LGTM

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/HistoryTests.swift (hunk 2 above)
Comment (clarmso) [reply firefox-ios/firefox-ios-tests/Tests/XCUITests/HistoryTests.swift:535]: The deleted lines looks spurious to me after checking with TestRail. :) LGTM.

**Pull request merged**